core/controllers/topics_and_skills_dashboard.py

```diff
@@ -167,21 +167,22 @@ def get(self):
 
 
 class SkillAssignmentsHandler(base.BaseHandler):
-    """Retrieving information about which topics contain the given skill."""
+    """Provides information about which topics contain the given skill."""
 
     GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
 
     @acl_decorators.can_access_topics_and_skills_dashboard
     def get(self, skill_id):
         """Handles GET requests."""
-        assigned_topics = skill_services.get_all_topics_assigned_to_skill(
+        assigned_topics_dict = skill_services.get_all_topics_assigned_to_skill(
```

Comment:
The naming here is getting very confusing: assigned_topics_dict, assigned_topic_dicts are too similar. In particular this makes line 185 very confusing.

Could you instead use have get_all_topics_assigned_to_skill return a list of TopicAssignment objects? I think that would make things much simpler (you can call the variables assigned_topics and assigned_topic_dicts).

Make sure that, in line 185, the key name matches the variable name.

Reply:
Yeah, I had a similar comment about returning dicts from a function, but the `TopicAssignment` object doesn't have a topic name field, and since that is required, I thought it would be fine, but yeah, the names are getting a bit confusing.
Maybe add topic name to that object and return a list?

Reply:
Yeah you guys are right, changed the dict to a list. 
Done.